### PR TITLE
 UI/Launcher: Modal button label customization

### PR DESCRIPTION
**Pull request by matheuszych**
As of right now, you are unable to set the labels for the buttons inside the `Launcher` modal. The default labels **Cancel** & **Save** are not always suitable for every purpose. Therefore I propose this change to make the labels adjustable if needed. If no label is provided, the default value **Cancel** or **Save** will be applied accordingly.

### CODE REVIEW

**Comment by dsstrassner** [comment]
hi @thibsy, could you estimate a response to this PR? I my understanding, we need this for the feature in PR #6298.

**Comment by thibsy** [comment]
Hi @Dennis1993, sorry this was not on my radar since @klees has been assigned, but I'll try to look into it this week. Please note that the interface part is missing, which also needs to pass JF. However, since this is a trivial change and the FR is probably for ILIAS 9, we can announce this change after the PR has been merged.

**Comment by dsstrassner** [comment]
@thibsy thanks for your quick response. I asked you, because you  set the labels.

**Review by thibsy — APPROVED** [review]
Hi @matheuszych

Thanks a lot for your contribution to the UI components!

I am aware this functionality is linked to a feature scheduled for ILIAS 9, which is why we cannot wait until next JF to announce this change. Luckily the changes are minimal and all LGTM, which allows us to announce it afterwards. I will therefore merge this if the checks pass and label it for next JF.

Kind regards,
@thibsy

**Comment by dsstrassner** [comment]
Thanks a bunch @thibsy!

**Comment by matthiaskunkel** [comment]
Will the modified labels become part of the language files? Or are they stored in another way?

**Comment by klees** [comment]
Hi @matthiaskunkel,

this introduces the possibility to set according labels of the modals, it does not introduce actual values they are set to. These values need to be determined by the component that is using the launcher (in this case: the T&A) and I absolutely expect the component to use the language files. The default values ("save" and "cancel") that the UI framework uses are in fact read from the lang-files.

Hope this helps!

**Comment by matthiaskunkel** [comment]
Thanks @klees for this clarification.

**Comment by matthiaskunkel** [comment]
Jour Fixe, 30 OCT 2023: We highly appreciate this suggestion and accept the PR for Trunk.

**Pull request merged**
